47 improve and test kni packages

bruno-at-orange opened the pull request:
- Improve test utilities
- Improve KNI test 
- fix #51

bruno-at-orange committed:
Add test-kni action

This action use the repo https://github.com/KhiopsML/KhiopsNativeInterface-tutorial
Use test-kni on deb and rpm packages

bruno-at-orange committed:
Modify PLMPI and mpi libraries dependencies

The purpose is to avoid binaries who don't need mpi to be dependant of it.

bruno-at-orange committed:
Remove kni-doc package

Remove all samples data of kni
Update install.cmake to handle this

bruno-at-orange committed:
add workflow to update the kni tutorial repository

bruno-at-orange committed:
Add unit test for KNI, improve utils test utilities

Add new test binary:
	- move sources frome KNItransfer
	- Add test reference

Improve test utilities:
 - new Utils library
 - remove "inline" for FileCompareForTest
 - improve stdout redirection (works for cout and printf)

bruno-at-orange committed:
Remove FULL option in CMake (and more)

Move back KNIRecodeFiles to KNITransfer
Improve Comparaison for unit test

bruno-at-orange committed:
bug fix: in KNI drivers are not registered several times

Pour empêcher que les drivers soient charger plusieurs fois, ajout de la
méthode IsExternalDriversRegistered() et intégration dans KhiopsNativeInterface.cpp

bruno-at-orange committed:
Unit Test : improve comparison method

Amélioration de la comparaison avec le fichier de reference en prenant en
compte le répertoire des données (remplacé par @ROOT_DIR@ dans les fichiers
de référence) et le répertoire temporaire (remplacé par @TMP_DIR@)

bruno-at-orange committed:
Unit Test: update KNITest and its reference file

bruno-at-orange committed:
Removing the last separator in GetSystemTmpDir()

Sur apple, le chemin du repertoire temporaire se termine par un '/'
ce qui n'est pas le cas sur windows et linux. Pour unifier, on supprime
ce dernier caractere

bruno-at-orange committed:
TestService: refactoring of the macro + windows porting

- La macro de test est passée en code normal dans le fichier cpp.
C'est plus simple à maintenir et ça factorise le code.
- Portage vers windows de la redirection du stdout
- Remplacement des \ en / dans lors de la comparaison des fichiers
de références (en plus de TMP_DIR et RROT_DIR)

bruno-at-orange committed:
Add Adult to LearningTest.

Used in KNITest

marcboulle reviewed (dismissed):
Voir les commentaires

bruno-at-orange committed:
Add script to kni-tutorial and test-kni

add script to get the same script in the tutorial and in the action test-kni

bruno-at-orange committed:
Remove the MPI dependency of KNI

Update CMakeLists.txt to add limpich.so dependency only to binaries
who actually need it (e.g. MODL KNITransfer and PLTest)

bruno-at-orange committed:
fix memory limit check in KNI

folmos-at-orange reviewed (changes requested):
Mostly minor stuff and some questions.

Also I think some commits should be squashed (there is one WIP) and there is a weird merge commit from `dev` that should not be there.

I'd also try to keep the commits titles in english (with the body we can allow some french 😀 )

folmos-at-orange reviewed (approved):
LGMT
